socorro/scripts/fetch_crashids.py

```diff
@@ -29,17 +32,77 @@
 
 HOST = 'https://crash-stats.mozilla.com'
 
+MAX_PAGE = 1000
 
-def fetch_crashids(params):
+
+@total_ordering
+class Infinity(object):
+    """Infinity is greater than anything else except other Infinities
+
+    NOTE(willkg): There are multiple infinities and not all infinities are equal, so what we're
+    doing here is wrong, but it's helpful. We can rename it if someone gets really annoyed.
+
+    """
+    def __eq__(self, obj):
+        return isinstance(obj, Infinity)
+
+    def __lt__(self, obj):
+        return False
+
+    def __repr__(self):
+        return 'Infinity'
+
+
+# For our purposes, there is only one infinity
+INFINITY = Infinity()
+
+
+def fetch_crashids(params, num_results):
+    """Generator that returns crash ids
+
+    :arg dict params: dict of super search parameters to base the query on
+    :arg varies num: number of results to get or INFINITY
+
+    :returns: generator of crash ids
+
+    """
     url = HOST + '/api/SuperSearch/'
 
-    resp = requests.get(url, params)
-    if resp.status_code == 200:
+    # Set up retrying on 429s
+    retries = Retry(total=32, backoff_factor=1, status_forcelist=[429])
+    session = requests.Session()
+    session.mount('http://', HTTPAdapter(max_retries=retries))
+    session.mount('https://', HTTPAdapter(max_retries=retries))
+
+    # Set up first page
+    params['_results_offset'] = 0
+    params['_results_number'] = min(MAX_PAGE, num_results)
+
+    # Fetch pages of crash ids until we've gotten as many as we want or there aren't any more to get
+    crashids_count = 0
+    while True:
+        resp = session.get(url, params=params)
+        if resp.status_code != 200:
+            raise Exception('Bad response: %s %s' % (resp.status_code, resp.content))
+
         hits = resp.json()['hits']
-        crashids = [hit['uuid'] for hit in hits]
-        return crashids
 
-    raise Exception('Bad response: %s %s' % (resp.status_code, resp.content))
+        for hit in hits:
+            crashids_count += 1
+            yield hit['uuid']
+
+            # If we've gotten as many crashids as we need, we return
+            if crashids_count >= num_results:
```

Comment:
This works for non-all cases with normal comparison of two integers.

This also works with the `all` case because `num_results` will be `INFINITY` and since nothing is bigger than `INFINITY` (as we've coded it), this will always return `False` and allow the loop to continue.